### ISSUE TITLE
MSI installer failing to detect WSL on Windows 10

### ISSUE THREAD

**Issue by Paskeskum** [Difegue/LANraragi#820]
**LRR Version and OS**
LRR 0.8.7 - .msi installer
Windows 10 Edu 20H2

**Bug Details**
When attempting to upgrade from 0.8.7, the installer claims that WSL is not installed. It is in fact installed, and prior versions all installed, upgraded and worked fine; nothing about the system, WSL or LRR changed since upgrading to 0.8.7. Currently installed version 0.8.7 still launches and operates as it should. I assume the issue has something to do with changes to WSL detection in installer for versions 0.8.8 and beyond.

**Matching Logs**
Full installer log - https://pastebin.com/JuM1JACX

**Screenshots**
![installer](https://github.com/Difegue/LANraragi/assets/16615962/c5a698c9-46c9-4c74-b7b2-8cc857f3d42d)
![wslconfig](https://github.com/Difegue/LANraragi/assets/16615962/fb3f018a-1dc2-4e8f-92c6-8842ba6df869)
![0 8 7 working](https://github.com/Difegue/LANraragi/assets/16615962/42194fd4-3df4-4367-9988-cf5f14975687)

**Comment by Paskeskum** [comment]
I forgot to check the error details in the launcher, which confirmed that wsl --status failed. wsl --update doesn't work either. I assume it's because I have a relatively old WSL version as well as Windows itself. I installed the latest WSL kernel update (5.10.102.2) and nothing changed. Is there anything I can do here apart from installing a newer version of Windows?

![error-details](https://github.com/Difegue/LANraragi/assets/16615962/c5473cda-1a0a-414c-9e5c-0be7c303101f)

**Comment by Paskeskum** [comment]
I'm pretty sure I already have WSL2, since I could use --set-version to convert LRR to 2, and I updated the WSL kernel as well (https://www.catalog.update.microsoft.com/Search.aspx?q=wsl). I think the issue is with my old wsl.exe executable. Unless I'm wrong, the only way to update it would be with Windows updates, and I have disabled the whole Windows Update part of Windows. I found some updates myself that should do the job, but none of them installed successfully.
Actually, I tried "migrating" to a next major W10 release to get the needed updates, but apparently my current version has a well-known bug that makes it impossible to preserve user data when updating... And the only way to resolve it is by installing the update KB4586853, which also fails to install for me.

So basically, I think I need a newer version of wsl.exe that recognizes arguments such as --status, but I don't think I can get it for my system with crippled Windows Update. I'll have to stick to LRR 0.8.7 for now, and unless I figure something else out, I will install a newer version of Windows to resolve the issue at some point.

**Comment by Difegue** [comment]
Hmm, I thought WSL2 always brought the new wsl.exe with it, but I guess not..  
You could also try installing the serviced-out-of-OS WSL from the MS Store if that's available to you. (Considering you've disabled WU, I'm guessing you probably killed the store alongside it but it doesn't hurt to ask)

**Comment by Paskeskum** [comment]
You are correct about the disabled Microsoft Store functionality on my system, but the apps from the store can be downloaded in .MSIXbundle format and installed through powershell. That worked out, but this new version of WSL (installed in a separate folder for MS apps) actually requires Windows to be updated in order to operate, so it's still the same problem in the end. As it turns out, WSL version really is tied to Windows version. I've reenabled the Microsoft Update, but the update to 22H2 fails, manual installations of suitable downloaded KBxxxxxxx updates similarly fail. That's the case only for updates to Windows, updates to various other components work fine both through Microsoft Update and manually. 

At some point later I'll fully reinstall Windows and with the newer WSL version everything will be resolved. I'm pretty sure there is no other way, so I'm closing the issue now. In any case, I appreciate the help, thank you.
